extensions/jsonb/runtime/src/main/java/io/quarkus/jsonb/JsonbProducer.java

```diff
@@ -13,6 +13,7 @@
 public class JsonbProducer {
 
     @Produces
+    @Singleton
     @DefaultBean
     public JsonbConfig jsonbConfig(Instance<JsonbConfigCustomizer> customizers) {
```

Comment:
`JsonbConfig` is not thread safe. So I don't think it's a good idea to share the instance. Do we really need to expose it as a bean?

Reply:
@geoand @loicmathieu ^ ?

Reply:
@mkouba ah, great catch!

By making it a bean users are able to supply there own configuration by means of a CDI producer, which you of course know is super helpful :).

So let's just remove the `@Singleton` here and add a comment on why we can't make it `@Singleton`, OK?

Reply:
> By making it a bean users are able to supply there own configuration by means of a CDI producer, which you of course know is super helpful :).

So an alternative could be to fire an event of type `JsonbConfig` when creating a `Jsonb` instance so that users can observe the config and modify it. But this would break the compatibility of course.

> So let's just remove the @Singleton here and add a comment on why we can't make it @Singleton, OK?

Sounds good enough. If there is a doc about overriding the `JsonbConfig` bean we should mention it there as well.

Reply:
@loicmathieu can you adjust your PR to only make `Jsonb` singleton.

Also I would add a comment for `JsonbConfig`: "JsonbConfig is not thread safe so it must not be made singleton."

Reply:
`must be @Dependent` maybe? It can't be `@ApplicationScoped` or any other scope that is shared.

Reply:
And also like @mkouba mentioned to mention the fact here: https://github.com/quarkusio/quarkus/blob/a1380eef3c107de2e1b2ddb4aa9ecb9ca6d0a10c/docs/src/main/asciidoc/rest-json.adoc#L162

Reply:
I switch to `@Dependent` even if it's the default scope so it's explicit that we want it.
I added the comment.

I also add the two scopes to the documentation so if someone provides their own beans they know which scope to add to them.

Reply:
Great, thanks!